tests/test_e2e_40_sdntrace.py

```diff
@@ -17,14 +18,27 @@ def setup_class(cls):
         cls.net.restart_kytos_clean()
         cls.net.wait_switches_connect()
         time.sleep(10)
-        circuit_id = cls.create_evc(400)
+        cls.circuit_id = cls.create_evc(400)
         time.sleep(10)
-        cls.circuit = cls.wait_until_evc_is_active(circuit_id)
+        cls.circuit = cls.wait_until_evc_is_active(cls.circuit_id)
 
     @classmethod
     def teardown_class(cls):
         cls.net.stop()
 
+    def teardown_method(cls):
```

Comment:
@gretelliz, this is already an improvement. This will ask that you follow a similar pattern that we have on other test suites, which is on `setup_class` only setup the underlyig topology and wait for the connections, but do not create any extra resource like this EVC, and then introduce a `setup_method` calling `self.net.start_controller(clean_config=True, enable_all=True)`, that way all teardown get simpler, because you'd have a guarantee that each test case will start with a clean config, and then provision each resource they need, and you won't need to delete any resource since you know the DB will get dropped before each test case starts.


Historically, this test suite used to provision this evc during `setup_class` since it was constantly reusing it, and it was simple enough, but when more coverage and other resources need to be created as you know it became cumbersome to maintain and the developer having the responsibility to also delete the resources. See what I mean? Let me know if it makes sense.

Reply:
Consequently, any `pytest` rerun attempt should also work without any conflict when an assert fails.